src/module_compilation.ml

```diff
@@ -58,13 +58,14 @@ let build_cm cctx ?sandbox ?(dynlink=true) ~dep_graphs ~cm_kind (m : Module.t) =
         | Cmi | Cmo -> other_targets
       in
       let dep_graph = Ml_kind.Dict.get dep_graphs ml_kind in
+      let opaque = CC.opaque cctx && ctx.version >= (4, 03, 0) in
```

Comment:
I think that this kind of test should be done in `Context`, like ` Context.supports_opaque_option`. There's a similar code path for `keep_locs`, if you agree with that I can extract the relevant functions in a separate PR.

Reply:
Yeah, feel free to do it post this PR. I don't have a strong preference where the check should be done but I suppose that diml should have a loc b/c I just kept things as is.

Reply:
Actually, this made me realize that we should turn off the option everywhere altogether if the OCaml version is below 4.03.0. This is not the only place we should be doing this.